Promote security considerations to the top (#127)
mxschmitt/action-tmate: README.md

    - uses: actions/checkout@v2
    - name: Setup tmate session
      uses: mxschmitt/action-tmate@v3

>       with:

Reviewer: Could we add a comment here saying that this requires a public SSH key to be added to the GitHub profile, and a direct link to the profile where that could be done for convenience?

Author: Can you make a proposal?

Reviewer 2: To this link I guess? https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account

Reviewer: I was thinking more about https://github.com/settings/ssh/new, but both links would work for me.

Reviewer: In addition to linking to https://github.com/settings/ssh/new, I would like to see in a comment that the key is needed to protect any secrets that are available in the workflow step, and that the `limit-access-to-actor` setting can be set to `false` for convenience if there is no risk for spilling secrets.

In my mind, this improvement is needed for this PR to be merged.

        limit-access-to-actor: true
```

To get the connection string, just open the `Checks` tab in your Pull Request and scroll to the bottom. There you can connect either directly per SSH or via a web based terminal.

![GitHub Checks tab](./docs/checks-tab.png "GitHub Checks tab")

## Security consideration: Use registered public SSH key(s)

By default *anybody* can connect to the tmate session.

> This can lead to security issues, for example if secrets are available in environment variables or in subsequent steps privileged code is executed.

Reviewer: Here, we need to inform users that secrets are _not_ made available by default, but that the risk comes from configuring `token` in the `action-tmate` Action or from adding `${{ secret.X }}` values to the `env` block, or from writing out secret values to disk in preceding workflow steps.

Author: for potential follow-uppers:
... or from configurations that are generated/downloaded/... by previous steps having access to secrets. Or from subsequent steps deploying signed code that has been produced including injected bits or from a whole variety of other reasons we did not think about right now, and probably the 90% of users who don't understand SSH keys will neither understand so we better stay on the safe side and try to be careful if in doubt.

> You should opt-in to install the public SSH keys [that you have registered with your GitHub profile](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account) to mitigate this risk.

Reviewer: We should mention the key `limit-access-to-actor` already here. Otherwise it's not clear that this setting has anything to do with the opt-in that this sentence talks about.

Author: Can you make a proposal?

Reviewer: Sadly, I do not have the time I would need to spend on this, and have to rely on you to iterate on this PR.

Author: ```suggestion
You should opt-in to install the public SSH keys [that you have registered with your GitHub profile](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account) and to set `limit-access-to-actor: true` to mitigate this risk.
```

> You should opt-in to install the public SSH keys [that you have registered with your GitHub profile](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account) to mitigate this risk.

Reviewer: Let's avoid the "should" here. The sentence will read much friendlier once it starts with "It is highly recommended to [...]".

Author: ```suggestion
It is highly recommended to opt-in to install the public SSH keys [that you have registered with your GitHub profile](https://docs.github.com/en/github/authenticating-to-github/adding-a-new-ssh-key-to-your-github-account) to mitigate this risk.
```

```yaml
name: CI
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Setup tmate session
      uses: mxschmitt/action-tmate@v3
      with:
        limit-access-to-actor: true
```

> If the registered public SSH key is not your default private SSH key, you will need to specify the path manually, like so: `ssh -i <path-to-key> <tmate-connection-string>`.

Reviewer: "... even if the workflow run suggests the command without the `-i` option"

Author: ```suggestion
If the registered public SSH key is not your default private SSH key, you will need to specify the path manually, like so: `ssh -i <path-to-key> <tmate-connection-string>  even if the workflow run suggests the command without the `-i` option`.
```
